src/regex/pcre2.cr

```diff
@@ -9,6 +9,20 @@ module Regex::PCRE2
     @re = PCRE2.compile(source, pcre2_options(options) | LibPCRE2::UTF | LibPCRE2::NO_UTF_CHECK | LibPCRE2::DUPNAMES | LibPCRE2::UCP) do |error_message|
       raise ArgumentError.new(error_message)
     end
+
+    jit_compile
+  end
+
+  private def jit_compile : Nil
+    ret = LibPCRE2.jit_compile(@re, LibPCRE2::JIT_COMPLETE)
+    if ret < 0
+      case error = LibPCRE2::Error.new(ret)
+      when .jit_badoption?
+        # okay
+      else
+        raise ArgumentError.new("Regex JIT compile error: #{error}")
```

Comment:
Isn't this a bit too harsh? Maybe without compilation the application might still work, even if slow. This said, I'm not sure how could we let the user know that something went wrong.

Reply:
I'm not sure what errors can actually reasonably appear here. When JIT compilation is not supported, that's indicated by `ERROR_JIT_BADOPTION` and handled in the previous branch (we ignore that).
Anything else, I don't know. But I would probably expect that would really be more fatal errors where you may not be able to just continue on.
A more plausible point for ignoring errors might be when JIT stack allocation fails (`jit_stack_create`).

Reply:
`PCRE2_ERROR_NOMEMORY` is another one we might want to not raise. I wonder if it makes sense to just ignore any error and have a `@compiled : Bool` to know if it was compiled or not